### ISSUE TITLE
Remove Boost

### ISSUE THREAD

**Issue by robertodr** [dev-cafe/libgetkw#6]
Boost is a heavyweight dependency and is only used to provide `boost::any`. We should find a way to pull the same trick without having to pull in the behemoth.

**Comment by bast** [comment]
Yes please!

**Comment by robertodr** [comment]
I'm not familiar with type erasure (the thing Boost Any does) so I need to read up a bit. I am sure it's removable, we don't need a fully general type.

**Comment by stigrj** [comment]
Can perhaps `std::any` replace `boost::any`? Requires `C++17` though.

**Comment by robertodr** [comment]
I don't think we are ready for that quite yet... C++17 is still too cutting edge. Apologies from cutting you off from the discussion @bast and I had about Getkw. The current plan is to remove the compiled language bindings completely and have Getkw as just a Python module. See [here](https://dev-cafe.github.io/2018/09/14/getkw-01/) and [here](https://dev-cafe.github.io/2018/09/25/getkw-02/) for context and initial work. Mostly we'd want to keep the use as is now, with the difference that the parsed file is in JSON format (or YAML or anything else standard, really) The client (Python, C++, Fortran) will have to implement a JSON reader and there are many lightweight (header-only even for C++!) that do this already.

**Comment by bast** [comment]
Sorry from me for not communicating the blog post work to Stig. It would be great to get your input. I plan to send comments/PRs soon.

**Comment by stigrj** [comment]
No worries. It's great that things are happening :slightly_smiling_face:

**Comment by robertodr** [comment]
I'll make sure to ping you next time I write a post, so you can give feedback :smile_cat:
